keptn/keptn
# feat(installer): Add initcontainers to prevent installation errors

## This PR
- Adds initContainers to prevent `Error`s and `CrashLoopBackOff`s during a Keptninstallation.


### Related Issues
Fixes #7580

### Notes
This is my "best guess" as to the order of the dependencies and probably needs adjusting.

### Follow-up Tasks
This is a temporary fix and should be solved with other methods (like improved readiness probes and health checks).

### How to test
Install a version without this PR and you'll see the pods fall into `Error` and `CrashLoopBackOff` states until everything settles down. This is due to dependencies between the Keptn pods.

Install this version and initcontainers should prevent `Error`s and `CrashLoopBackOff`s.

Green integration test run: https://github.com/keptn/keptn/actions/runs/3264768834
